webnn/resources/test_data/cast.json

```diff
@@ -0,0 +1,3040 @@
+{
+  "tests": [
+    {
+      "name": "cast float32 0D tensor to int32",
+      "inputs": {
+        "input": {
+          "shape": [],
+          "data": [
+            84.77753448486328
+          ],
+          "type": "float32"
+        }
+      },
+      "type": "int32",
+      "expected": {
+        "name": "output",
+        "shape": [],
+        "data": [
+          84
+        ],
+        "type": "int32"
+      }
+    },
+    {
+      "name": "cast float32 1D tensor to int32",
+      "inputs": {
+        "input": {
+          "shape": [24],
+          "data": [
+            102.1578369140625,
```

Comment:
We should include some negative values too, one to test truncating down toward negative infinity, one toward zero, and half decimal. e.g. [-4.9, -5.2, 6.5]